smolyay/smolyak.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import scipy.special
+
+
+class IndexGrid:
+    """Generate index of grid points.
+
+    Attributes
+    ----------
+    dimension : int
+        Number of independent variables.
+    exactness : int
+        Level of exactness of the approximation.
+    index_per_level : list
+        Number of indexes at each level
+    """
+
+    def __init__(self, dimension, exactness, index_per_level):
+        self.dimension = dimension
+        self.exactness = exactness
+        self.index_per_level = index_per_level
+
+        if (type(index_per_level) == tuple or
+                type(index_per_level) == np.ndarray):
+            index_per_level = list(index_per_level)
+
+        if len(index_per_level) != exactness + 1:
+            raise IndexError(
+                "index_per_level must be an array with a length of {}"
+                .format(exactness + 1))
+
+    def level_indexes(self):
+        """Generate indexes of levels.
+
+        Returns
+        -------
+        level_indexes : numpy array
+            Unique indexes at each level.
+
+        """
+        # creating an array to store unique indexes in each level
+        level_indexes = np.zeros((self.exactness+1), dtype=list)
+
+        max_index = sum(self.index_per_level)
```

Comment:
Here the code requires that index_per_level cannot contain extra information. This part might be edited to avoid this.

Reply:
Addressed.